core/math/a_star.cpp

```diff
@@ -280,18 +280,23 @@ int AStar::get_closest_point(const Vector3 &p_point, bool p_include_disabled) co
 			continue; // Disabled points should not be considered.
 		}
 
+		// Keep the closest point's ID, and in case of multiple closest IDs,
+		// the smallest one (makes it deterministic).
 		real_t d = p_point.distance_squared_to((*it.value)->pos);
-		if (closest_id < 0 || d < closest_dist) {
+		int id = *(it.key);
+		if (d <= closest_dist) {
+			if (d == closest_dist && id > closest_id) { // Keep lowest ID.
+				continue;
+			}
 			closest_dist = d;
-			closest_id = *(it.key);
+			closest_id = id;
 		}
 	}
 
 	return closest_id;
 }
 
 Vector3 AStar::get_closest_position_in_segment(const Vector3 &p_point) const {
-	bool found = false;
 	real_t closest_dist = 1e20;
 	Vector3 closest_point;
```

Comment:
Strictly speaking this should be REAL_MAX or some such constant rather than an arbitrary value, I don't know if we have one (either FLT_MAX or double max depending on what real is defined as).

Reply:
Yeah but that's out of the scope of this PR IMO, `1e20` is used for "big enough distance" in quite a few places in the codebase. There's even:
```
core/math/geometry_3d.cpp
895:#define INF 1e20
```
:scream_cat:

Reply:
FYI:
```
$ rg 1e20
servers/physics_2d/shape_2d_sw.h
324:            r_max = -1e20;
325:            r_min = 1e20;

servers/physics_2d/body_2d_sw.h
310:            real_t least_depth = 1e20;

servers/physics_3d/shape_3d_sw.cpp
433:    float min_distance = 1e20;
533:    real_t min_d = 1e20;
764:    real_t min = 1e20;
843:    float min_distance = 1e20;
1145:   params.min_d = 1e20;

servers/physics_3d/space_3d_sw.cpp
479:    float min_distance = 1e20;

servers/physics_2d/physics_server_2d_sw.cpp
175:            real_t min_depth = 1e20;

scene/3d/baked_lightmap.cpp
483:            float min_d_dist = 1e20;

servers/physics_3d/physics_server_3d_sw.cpp
1415:           real_t min_depth = 1e20;

servers/physics_3d/body_3d_sw.h
349:            real_t least_depth = 1e20;

scene/3d/voxelizer.cpp
782:#define INF 1e20

scene/gui/gradient_edit.cpp
62:     float min_distance = 1e20;

servers/rendering/rendering_device.h
534:                    max_lod = 1e20; //something very large should do

scene/2d/polygon_2d.cpp
157:                            float highest_y = -1e20;

scene/main/viewport.cpp
562:                            Vector2 last_pos(1e20, 1e20);

servers/rendering/rendering_server_scene.cpp
1265:                           real_t z_max = -1e20;
1266:                           real_t z_min = 1e20;

servers/rendering/rasterizer_rd/rasterizer_effects_rd.cpp
1529:   sampler.max_lod = 1e20;

scene/resources/polygon_path_finder.cpp
140:            float closest_dist = 1e20;
164:            float closest_dist = 1e20;
492:    float closest_dist = 1e20;
511:    ERR_FAIL_COND_V(closest_dist == 1e20, Vector2());

scene/animation/animation_node_state_machine.cpp
229:            float least_cost = 1e20;
413:            float priority_best = 1e20;

scene/animation/animation_blend_space_2d.cpp
521:            float new_closest_dist = 1e20;

modules/csg/csg.cpp
1439:                           real_t min_a = 1e20, max_a = -1e20;
1440:                           real_t min_b = 1e20, max_b = -1e20;

modules/gdnavigation/nav_map.cpp
78:     float begin_d = 1e20;
79:     float end_d = 1e20;
212:                    end_d = 1e20;
389:    real_t closest_point_d = 1e20;
440:    real_t closest_point_d = 1e20;
466:    real_t closest_point_d = 1e20;
493:    real_t closest_point_d = 1e20;

modules/lightmapper_rd/lm_compute.glsl
150:                    float best_distance = 1e20;
497:    float min_d = 1e20;

modules/bullet/space_bullet.cpp
283:    btScalar out_distance = 1e20;

modules/bullet/space_bullet.h
200:            btScalar penetration_distance = 1e20; // Negative mean penetration

editor/animation_track_editor.cpp
2425:           float key_distance = 1e20;
2645:                   float key_distance = 1e20;
5220:                   float from_t = 1e20;
5221:                   float to_t = -1e20;
5222:                   float len = -1e20;

editor/node_3d_editor_gizmos.cpp
509:            float min_d = 1e20;
532:            min_d = 1e20;
611:            float cpd = 1e20;
825:    float min_d = 1e20;
1093:   float closest_dist = 1e20;
1094:   float closest_angle = 1e20;

editor/plugins/node_3d_editor_plugin.cpp
496:    float closest_dist = 1e20;
827:            float col_d = 1e20;
847:                    col_d = 1e20;
889:            float col_d = 1e20;
924:            float col_d = 1e20;
944:                    col_d = 1e20;

editor/plugins/polygon_2d_editor_plugin.cpp
571:                                    float closest_dist = 1e20;
641:                                    float closest_dist = 1e20;

editor/plugins/canvas_item_editor_plugin.cpp
1016:                   float minimum = 1e20;

editor/plugins/animation_state_machine_editor.cpp
188:            float closest_d = 1e20;
318:                    float best_d_x = 1e20;
319:                    float best_d_y = 1e20;

editor/plugins/path_3d_editor_plugin.cpp
323:                    float closest_d = 1e20;

core/math/face3.h
229:                    real_t minT = 1e20, maxT = -1e20;

core/math/triangle_mesh.cpp
384:    real_t d = 1e20;

core/math/geometry_3d.cpp
895:#define INF 1e20

core/math/a_star.cpp
276:    real_t closest_dist = 1e20;
300:    real_t closest_dist = 1e20;

core/math/geometry_2d.cpp
173:    real_t best_aspect = 1e20;

core/math/geometry_3d.h
357:            real_t min = -1e20, max = 1e20;

core/math/aabb.cpp
108:    real_t near = -1e20;
109:    real_t far = 1e20;

core/math/geometry_2d.h
329:            Vector2 further_away(-1e20, -1e20);
330:            Vector2 further_away_opposite(1e20, 1e20);
```